specs/IFrameMemoryUsage

```diff
@@ -0,0 +1,242 @@
+{
+    // Add frame created event for webViewFrames
+    _webViewFrames.Add(args.Frame);
+    args.Frame.FrameCreated += WebViewFrames_CreatedNestedIFrames;
+
+    args.Frame.Destroyed += (frameDestroyedSender, frameDestroyedArgs) =>
+    {
+        var frameToRemove = _webViewFrames.SingleOrDefault(r => r.IsDestroyed() == 1);
+        if (frameToRemove != null)
+            _webViewFrames.Remove(frameToRemove);
+    };
+}
+```
+
+# API Details
+## C++
+```cpp
+interface ICoreWebView2_18;
+interface ICoreWebView2Frame5;
+interface ICoreWebView2NestedFrameCreatedEventHandler;
+interface ICoreWebView2RenderProcessIdChangedEventHandler;
+interface ICoreWebView2FrameRenderProcessIdChangedEventHandler;
+
+/// Receives `FrameCreated` event.
+// MSOWNERS: wangsongjin@microsoft.com
+[uuid(c0e55260-a1bb-11ed-a8fc-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2NestedFrameCreatedEventHandler : IUnknown {
+  /// Provides the result for the iframe created event.
+  HRESULT Invoke([in] ICoreWebView2Frame* sender,
+                 [in] ICoreWebView2FrameCreatedEventArgs* args);
+}
+
+/// Receives `RenderProcessIdChanged` event.
+// MSOWNERS: wangsongjin@microsoft.com
+[uuid(012a41fe-a41f-11ed-a8fc-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2RenderProcessIdChangedEventHandler : IUnknown {
+  /// Provides the result for the process changed event.
+  /// No event args exist and the args parameter is set to null.
+  HRESULT Invoke([in] ICoreWebView2 * sender,
+                 [in] IUnknown* args);
+}
+
+/// Receives `ProcessIdChanged` event.
+// MSOWNERS: wangsongjin@microsoft.com
+[uuid(c1e0a3d2-a74b-11ed-afa1-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2FrameRenderProcessIdChangedEventHandler : IUnknown {
+  /// Provides the result for the process changed event.
+  HRESULT Invoke([in] ICoreWebView2Frame* sender,
+                 [in] IUnknown* args);
+}
+
+/// This is an extension of the ICoreWebView2Frame interface.
+// MSOWNERS: wangsongjin@microsoft.com
+[uuid(04baa798-a0e9-11ed-a8fc-0242ac120002), object, pointer_default(unique)]
+interface ICoreWebView2Frame5 : ICoreWebView2Frame4 {
+  /// Raised when a new iframe is created.
+  /// Handle this event to get access to ICoreWebView2Frame objects.
+  /// Use ICoreWebView2Frame.add_Destroyed to listen for when this iframe goes
+  /// away.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT add_FrameCreated(
+      [in] ICoreWebView2NestedFrameCreatedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with add_FrameCreated.
+  // MSOWNERS: wangsongjin@microsoft.com
+  HRESULT remove_FrameCreated([in] EventRegistrationToken token);
+
+  /// Get the render process id of the iframe.
+  // MSOWNERS: wangsongjin@microsoft.com
+  [propget] HRESULT RenderProcessId([out, retval] UINT32* value);

```

Comment:
No existing APIs use render process as naming. Agree!

Reply:
Looks like both show up already. Since we're already inconsistent we can go with the more accurate one renderer. COREWEBVIEW2_PROCESS_FAILED_KIND uses 'render' and COREWEBVIEW2_PROCESS_KIND uses 'render**er**'

Reply:
Renamed.